cli/flox-rust-sdk/src/flox.rs

```diff
@@ -253,15 +253,18 @@ pub enum FloxhubTokenError {
 #[derive(Debug, Clone)]
 pub struct Floxhub {
     base_url: Url,
+    git_url: Url,
     git_url_override: Option<Url>,
 }
 
 impl Floxhub {
-    pub fn new(base_url: Url) -> Self {
-        Floxhub {
+    pub fn new(base_url: Url) -> Result<Self, FloxhubError> {
+        let git_url = Self::derive_git_url(&base_url)?;
```

Comment:
this will fail if you set a different (syntactically incompatible) floxhub url, but it should not fail if you have an override -- i guess if you want the deriving in the constructor, you also need to take the override (which would _override_ also the derive error)

Reply:
Yeah I just reverted that...can revert the revert